services/device-registry-base/src/main/java/org/eclipse/hono/service/management/credentials/RPKSecret.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.service.management.credentials;
+
+import java.security.PublicKey;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+import java.util.Objects;
+
+import org.eclipse.hono.service.auth.ExternalJwtAuthTokenValidator;
+import org.eclipse.hono.util.CredentialsConstants;
+import org.eclipse.hono.util.RegistryManagementConstants;
+import org.eclipse.hono.util.Strings;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+/**
+ * This class encapsulates the secrets information for a raw public key credentials type.
+ */
+@RegisterForReflection
+@JsonInclude(value = JsonInclude.Include.NON_NULL)
+public final class RPKSecret extends CommonSecret {
+
+    @JsonProperty(RegistryManagementConstants.FIELD_SECRETS_KEY)
+    private byte[] key;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_SECRETS_ALGORITHM)
+    private String alg;
```

Comment:
can we rename this field to *algorithm* and adapt the setter and getter accordingly?

Reply:
sure, renamed